Ghiurutan V. Alexandru/Homework5/src/view/Gui.java

```diff
@@ -0,0 +1,164 @@
+		search.addActionListener(this);
+
+		operations.add(add);
+		operations.add(delete);
+		operations.add(search);
+		this.add(operations, BorderLayout.SOUTH);
+		this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		this.pack();
+		this.setVisible(true);
+	}
+
+	private void fillTable() {
+		if (dictionary.getTotalNrOfWords() != 0) {
+			Set<Entry<Word, String>> diction = dictionary.getAllWordsInDictionary();
+			rows = new Object[diction.size()][2];
+			Iterator<Entry<Word, String>> it = diction.iterator();
+			while (it.hasNext()) {
+				Entry<Word, String> newWord = it.next();
+				Object[] row = { newWord.getKey().getWord(), newWord.getValue() };
+				tableModel.addRow(row);
+			}
+		}
+	}
+
+	private void updateTable() {
+		tableModel = new DefaultTableModel(rows, columns);
+		tableModel.setRowCount(0);
+		table.setModel(tableModel);
+		Set<Entry<Word, String>> diction = dictionary.getAllWordsInDictionary();
+		rows = new Object[diction.size()][2];
+		Iterator<Entry<Word, String>> it = diction.iterator();
+		while (it.hasNext()) {
+			Entry<Word, String> newWord = it.next();
+			Object[] row = { newWord.getKey().getWord(), newWord.getValue() };
+			tableModel.addRow(row);
+		}
+	}
+
+	private void initializeNorthSection() {
+		tableModel = new DefaultTableModel(rows, columns);
+		table = new JTable();
+		table.setModel(tableModel);
+		table.setEnabled(false);
+		fillTable();
+		scrollPane = new JScrollPane(table);
+		this.add(scrollPane, BorderLayout.NORTH);
+	}
+
+	private String checkInputWord() {
+		String word = String.valueOf(
+				JOptionPane.showInputDialog(this, "Please enter the word.", "Input", JOptionPane.INFORMATION_MESSAGE))
+				.toLowerCase();
+		while (word.equals("")) {
+			JOptionPane.showMessageDialog(this, "You didn't enter a valid word.Please enter again more carefully.",
+					"ERROR", JOptionPane.ERROR_MESSAGE);
+			word = String.valueOf(JOptionPane.showInputDialog(this, "Please enter the word .", "Input",
+					JOptionPane.INFORMATION_MESSAGE)).toLowerCase();
+		}
+		return word;
+	}
+
+	private String checkInputDescription(String auxWord) {
+		String description = String.valueOf(JOptionPane.showInputDialog(this,
+				"Enter the description of the word: " + auxWord + ", that you want in dictionary.", "Input",
+				JOptionPane.INFORMATION_MESSAGE)).toLowerCase();
+		while (description.equals("")) {
+			JOptionPane.showMessageDialog(this, "You didn't enter a valid word.Please enter again more carefully.",
+					"ERROR", JOptionPane.ERROR_MESSAGE);
+			description = String.valueOf(JOptionPane.showInputDialog(this,
+					"Enter the description of the word: " + auxWord + ", that you want in dictionary.", "Input",
+					JOptionPane.INFORMATION_MESSAGE)).toLowerCase();
+		}
+		return description;
+	}
+
+	private void addWords(Word newWord) {
+		String description = checkInputDescription(newWord.getWord());
+		dictionary.addWord(newWord, description);
+		description = description.replaceAll("[^A-Za-z-]", " ");
```

Comment:
I ensure here that I don't have words with special characters or too much free spaces.